pkg/controller/dataimportcron-controller.go

```diff
@@ -557,22 +634,161 @@ func (r *DataImportCronReconciler) createImportDataVolume(ctx context.Context, d
 	if err != nil {
 		return err
 	}
-	// If PVC exists don't create DV
+	dataImportCron.Status.CurrentImports = []cdiv1.ImportStatus{{DataVolumeName: dvName, Digest: digest}}
+
 	pvc := &corev1.PersistentVolumeClaim{}
 	if err = r.client.Get(ctx, types.NamespacedName{Namespace: dataImportCron.Namespace, Name: dvName}, pvc); err != nil {
 		if !k8serrors.IsNotFound(err) {
 			return err
 		}
-		dv := r.newSourceDataVolume(dataImportCron, dvName)
-		if err := r.client.Create(ctx, dv); err != nil && !k8serrors.IsAlreadyExists(err) {
+	} else {
+		if err := r.updateSourceObject(ctx, dataImportCron, pvc); err != nil {
+			return err
+		}
+		// If PVC exists don't create DV
+		return nil
+	}
+
+	snapshot := &snapshotv1.VolumeSnapshot{}
+	if err = r.client.Get(ctx, types.NamespacedName{Namespace: dataImportCron.Namespace, Name: dvName}, snapshot); err != nil {
+		if !k8serrors.IsNotFound(err) && !meta.IsNoMatchError(err) {
 			return err
 		}
 	} else {
-		if err := r.updatePvc(ctx, dataImportCron, pvc); err != nil {
+		if err := r.updateSourceObject(ctx, dataImportCron, snapshot); err != nil {
 			return err
 		}
+		// If snapshot exists don't create DV
+		return nil
 	}
```

Comment:
Similar handling for pvc/snapshot, so either use a helper func or iterate over them.

Reply:
done